### PR TITLE
Move responsibility for dealing with validation to schemas

### PR DESCRIPTION
**Pull request by Drvanon**
Because of the gridtlib, it will be necessary to move any interaction with the models out of the resources and into the controllers. The interactions of the database with the schemas can then much more nicely be isolated into the controllers, so that the schema simply has to call upon them.

### CODE REVIEW

File: web/gridt/resources/movements.py [hunk 1]
```diff
@@ -9,40 +9,30 @@
...
-        existing_movement = Movement.find_by_name(res["name"])
-        if existing_movement:
-            return (
-                {
-                    "message": "Could not create movement, because movement name is already in use."
-                },
-                400,
-            )
+        current_identity = User.query.get(get_jwt_identity())
+        data = schema_loader(self.schema, request.get_json())
```
Comment (marickmanrho) [inline web/gridt/resources/movements.py:29]: So much cleaner!

**Review by marickmanrho — COMMENTED** [review]
Looks very nice. It's quite tough to determine if there are any loopholes in your logic. So for now I just review this as inconclusive. If you are certain about your work, feel free to merge. At least the PR looks good and does the job as far as I can see.

File: web/gridt/resources/movements.py (hunk 1 above)
Comment (Drvanon) [reply web/gridt/resources/movements.py:29]: I am planning on making this a part of a new subclass of Resource, which probably would play together with the controller object as well.

File: web/gridt/resources/user.py [hunk 2]
```diff
@@ -25,16 +25,20 @@
...
+    schema = BioSchema()
+
+    @jwt_required
     def put(self):
-        schema = BioSchema()
+        current_identity = User.query.get(get_jwt_identity())
 
         data = request.get_json()
         bio = None
         try:
```
Comment (Drvanon) [inline web/gridt/resources/user.py:40]: Should make use of schema loader

File: web/gridt/resources/user.py [hunk 3]
```diff
@@ -47,8 +51,10 @@ def put(self):
...
     schema = ChangePasswordSchema()
 
-    @jwt_required()
+    @jwt_required
     def post(self):
+        current_identity = User.query.get(get_jwt_identity())
+
         try:
             res = self.schema.load(request.get_json())
         except ValidationError as error:
```
Comment (Drvanon) [inline web/gridt/resources/user.py:60]: Use schema_loader

**Pull request closed without merging**